File: ocaml/difference-of-squares/difference_of_squares.ml
let square x = x * x
let natural_numbers number = List.init number (fun x -> x + 1)
let sum numbers = List.fold_left (+) 0 numbers

let square_of_sum number =  natural_numbers number |> sum |> square
let sum_of_squares number = natural_numbers number |> List.map square |> sum
let difference_of_squares number = square_of_sum number - sum_of_squares number 

File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number)
{
    return number * (number + 1) * (2 * number + 1) / 6;
}

unsigned int square_of_sum(unsigned int number)
{
    int sum = number * (number + 1) / 2;
    return sum * sum;
}

unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}
